packages/terra-date-picker/src/DateUtil.js

```diff
@@ -30,6 +30,40 @@ class DateUtil {
     return momentDates.length > 0 ? momentDates : dates;
   }
 
+  // Checks if a given date is out of the range between the start and end dates.
+  static isDateOutOfRange(sourceDate, startDate, endDate) {
+    if (!sourceDate || !sourceDate.isValid() ||
+      !startDate || !startDate.isValid() ||
+      !endDate || !endDate.isValid()) {
+      return false;
+    }
+
+    if (sourceDate.isSameOrAfter(startDate) && sourceDate.isSameOrBefore(endDate)) {
```

Comment:
Are .isSameOrAfter & .isSameOrBefore moment functions?

Reply:
Yes, these are moment [functions](http://momentjs.com/docs/#/query/is-same-or-before/).